### ISSUE TITLE
Mac: issue with Catalina/zip files

### ISSUE THREAD

**Issue by D3nnis3n** [StrangeLoopGames/EcoIssues#13867]
Eco doesn't work on MacOS Catalina. Users report they got the warning in prior versions that MacOS will no longer support 32-Bit Software in the future when starting, with Catalina Eco ceased to start.

Not sure if that is actually the cause, i only have what they report to me.

**Comment by redwyre** [comment]
The overall app is 64 bit but there might be a stray plugin of something that's setting it off. A player log from a user might help.

Another cause could be the use of OpenGL, I'll try testing a version with OpenGL vs Metal.

**Comment by Nothing4You** [comment]
<img width="532" alt="image" src="https://user-images.githubusercontent.com/3694534/70102999-1eddfb80-163a-11ea-8fc1-f993bcc30c9b.png">

```
Dec  4 02:02:19 macos com.apple.xpc.launchd[1] (com.apple.xpc.launchd.oneshot.0x1000003d.Eco[45767]): Could not find and/or execute program specified by service: 13: Permission denied: /Applications/Eco.app/Contents/MacOS/Eco
Dec  4 02:02:19 macos com.apple.xpc.launchd[1] (com.apple.xpc.launchd.oneshot.0x1000003d.Eco[45767]): Service exited with abnormal code: 78
```

I believe the game package is just broken.
`Eco.app/Contents/MacOS/Eco` should be executable I believe, which is likely lost somewhere in the creation process. Since the application is signed after this it's also not possible to fix this by running `chmod +x`

**Comment by redwyre** [comment]
I found a magic flag in steam I've set to say it's 64bit, maybe that will fix launching from steam...

**Comment by enl** [comment]
I can confirm that the problem is with packaging. If I unpack zip archive with "Archive Utility" (default option) it results in broken app.

If I right click on archive and open it with "Unarchiver" app (another tool to work with zip archives) it successfully unpacks runnable application. 

In addition, Steam version starts up fine as well.

**Comment by redwyre** [comment]
so steam is fine, but zip is broken?

**Comment by enl** [comment]
@redwyre yup.

**Comment by redwyre** [comment]
Would you be able to compare the unzipped folders from "Archive Utility" and "Unarchiver"?

**Comment by enl** [comment]
```
% diff -rq Eco.app Eco_unarchiver.app
```

Gives empty results.

**Comment by redwyre** [comment]
hmmm
what about if you run `ls -l` on each directory and redirect to a file, and compare the files?

**Comment by Uniphix** [comment]
I can confirm that when using OSX Archive Utility it does not unpackage correctly as well.  I downloaded the Unarchiver and it works just fine.

**Comment by redwyre** [comment]
So for the current issue, what I'm expecting it to be is the zips are created on windows, and macOS expects the execute bit set on the executable. Steam automatically does this.
Could also be an issue with the plist file, not sure what steam does here.

**Comment by redwyre** [comment]
I have a test zip available at https://eco-releases-staging.s3-us-west-2.amazonaws.com/EcoOSX_v0.9.2.4-beta-permissions_test.zip

**Comment by redwyre** [comment]
this isn't fixed yet

**Comment by redwyre** [comment]
I have checked in a ZipAttributes project that modifies the zip attributes, it's still not working so still requires some tinkering

**Comment by mirasrael** [comment]
Spend one day on fix or investigation. If it isn't something simple to fix then return to Icebox.

**Comment by thetestgame** [comment]
After a quick test it the issue seems to be that during the zip process the executable permission is being stripped off the file located at ``Eco.App/Contents/MacOS/Eco``. Manually adding the permission back with ``chmod +x ./EcoApp/Contents/MacOs/Eco`` resolves this issue. It might be better to use something like a tar file or dmg for Mac users rather then a zip file.

**Comment by WeaselDog** [comment]
Fixed in 0.9.5
